src/App.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import Button from 'react-bootstrap/Button'
+import {
+  clickNumber,
+  clickOp,
+  clickClear,
+  clickSubmit
+} from './redux/actions'
+
+function fireAction(props,id) {
+  const isNumerical = !Number.isNaN(parseInt(id)) || id === "."
+  const isOperator = id === "+" || id === "-" || id === "*" || id === "/"
+  const isClear = id === "C"
+  const isSubmit = id === "="
+
+  if(isNumerical) {
+    props.clickNumber(id)
+  }
+  else if(isOperator) {
+    props.clickOp(id)
+  }
+  else if(isClear) {
+    props.clickClear()
+  }
+  else if(isSubmit) {
+    props.clickSubmit()
+  }
+}
+
+function App(props) {
+  return (
+    <>
+      <div className="app-container">
+        <div className="calculator" onClick={(evt)=>fireAction(props,evt.target.id)}>
```

Comment:
To keep this section clean, you could create the number buttons with a loop:

```
let buttons = [];
for (let i = 0; i < 10; i++) {
buttons.push(<Button id={i} key={i} style={{gridArea: `button${i}`})>{i}</Button>
}
```

Reply:
Haha, I should have seen this when I used my editor's regex find and replace to create these lines... if I can programmatically write the lines of code, I can programmatically inject them from a single template!